client/maps/map-image.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+
+import ImageIcon from '../icons/material/baseline-image-24px.svg'
+
+import { grey800 } from '../styles/colors'
+import { Subheading } from '../styles/typography'
+
+const ImgElement = styled.img`
+  display: block;
+  width: 100%;
+  object-fit: cover;
+`
+
+const NoImage = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+  min-height: 220px;
+  background-color: ${grey800};
+
+  & > svg {
+    width: 90px;
+    height: 90px;
+    opacity: 0.5;
+  }
+`
+
+export default class MapImage extends React.Component {
+  static propTypes = {
+    map: PropTypes.object.isRequired,
+    size: PropTypes.number,
+    showNotAvailableText: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    size: 256,
+  }
+
+  render() {
+    const { map, size, showNotAvailableText } = this.props
+    const srcSet = `
+      ${map.image256Url} 256w,
+      ${map.image512Url} 512w,
+      ${map.image1024Url} 1024w,
+      ${map.image2048Url} 2048w
+    `
+
+    // TODO(2Pac): Actually check if the image URL is available instead of just checking it's set.
+    return (
+      <>
```

Comment:
This component now returns either:
- An image element that is styleable
- A div that is not at all styleable

That doesn't feel like a good API to me, ideally this component would have the same degree of styleability regardless of whether the image is present. It's going to get *especially* bad if you start handling 404s, where that can change on the fly. Since you feel like the NoImage case needs to have an option for text anyway, maybe a better option is to allow the caller to provide the element to use if the image isn't available? That'd fix the issues I noted above, and also remove the need for that configuration option.

Reply:
Yeah, unstyleability of the NoImage case was annoying me as well :d
Do you think this component should still have some kind of a default way to display a NoImage case?

Reply:
I think pulling out the current one into a component and making that the default if it's not specified would be fine